sktime/transformations/panel/window_summarizer.py

```diff
@@ -0,0 +1,268 @@
+#!/usr/bin/env python3 -u
+# -*- coding: utf-8 -*-
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+"""Extract features across (shifted) windows from provided series."""
+
+__author__ = ["Daniel Bartling"]
+__all__ = ["LaggedWindowSummarizer"]
+
+import pandas as pd
+from joblib import Parallel, delayed
+
+from sktime.transformations.base import _PanelToTabularTransformer
+
+# List of native pandas rolling window function.
+# danbartl: investivate different engines for pandas
+pd_rolling = [
+    "sum",
+    "mean",
+    "median",
+    "std",
+    "var",
+    "kurt",
+    "min",
+    "max",
+    "corr",
+    "cov",
+    "skew",
+    "sem",
+]
+
+
+def _window_feature(
+    Z,
+    name=None,
+    win_summarizer=None,
+    window=None,
+):
+    """Compute window features and lag.
+
+    Apply functions passed over a certain window
+    of past observations, e.g. the mean of a window of length 7 days, lagged by 14 days.
+
+    y: either pandas.core.groupby.generic.SeriesGroupBy
+        Object create by grouping across groupBy columns.
+    name : str, base string of the derived features
+           final name will also contain
+           window shift and window length.
+    win_summarizer: either str corresponding to native
+    implemented pandas function, currently
+            * "sum",
+            * "mean",
+            * "median",
+            * "std",
+            * "var",
+            * "kurt",
+            * "min",
+            * "max",
+            * "corr",
+            * "cov",
+            * "skew",
+            * "sem"
+         or function definition. See for the native functions also
+         https://pandas.pydata.org/docs/reference/window.html.
+    window: list of integers
+        Contains values for window shift and window length.
+    """
+    win = window[0]
+    shift = window[1] + 1
+
+    if win_summarizer in pd_rolling:
+        if isinstance(Z.index, pd.MultiIndex):
+            feat = getattr(Z.shift(shift).rolling(win), win_summarizer)()
+        else:
+            feat = pd.DataFrame(Z).apply(
+                lambda x: getattr(x.shift(shift).rolling(win), win_summarizer)()
+            )
+    else:
+        feat = Z.shift(shift)
+        if isinstance(Z.index, pd.MultiIndex) and callable(win_summarizer):
+            feat = feat.rolling(win).apply(win_summarizer, raw=True)
+        elif not isinstance(Z.index, pd.MultiIndex) and callable(win_summarizer):
+            feat = feat.apply(lambda x: x.rolling(win).apply(win_summarizer, raw=True))
+
+    if isinstance(feat, pd.Series):
+        feat = pd.DataFrame(feat)
+
+    feat.rename(
+        columns={
+            feat.columns[0]: name + "_" + "_".join([str(item) for item in window])
+        },
+        inplace=True,
+    )
+
+    return feat
+
+
+class _LaggedWindowExtractor(_PanelToTabularTransformer):
```

Comment:
why are there two classes? Seems one adds `_fit`, the oder one `_transform`, so why is not both in one single class?